bootstrap/main.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import pathlib
+import shutil
+import sys
+import time
+from typing import Any
+
+import docker
+
+client = docker.from_env()
+
+default_file_path = pathlib.Path("/startup.json.default")
+config_file_path = pathlib.Path("/config/startup.json")
+
+CORE_CONTAINER_NAME = "companion_core"
+
+
+def ensure_dir(file_path: str) -> None:
+    """Makes sure that the file path exists
+
+    Args:
+        file_path (str): path to check/make
+    """
+    directory = os.path.dirname(file_path)
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def stop_old_container() -> None:
+    """ Looks for an old core container and stops it if found"""
+    try:
+        old_container = client.containers.get(CORE_CONTAINER_NAME)
+        old_container.stop()
+        old_container.remove()
+    except Exception:
+        pass
+
+
+def overwrite_config_file_with_defaults() -> None:
+    """Overwrites the config file with the default configuration"""
+    shutil.copy(default_file_path, config_file_path)
+
+
+def read_config_file() -> Any:
+    """Tries to read the config file
+
+    Returns:
+        Any: Json data of startup.json
+    """
+
+    # Tries to open the current file
+    try:
+        with open("/config/startup.json") as config_file:
+            return json.load(config_file)
+    except Exception:
+        # Copy defaults over and read again
+        shutil.copy(default_file_path, config_file_path)
+        with open("/config/startup.json") as config_file:
+            return json.load(config_file)
+
+
+def start_core() -> None:
+    """Loads core settings and launches the core docker. Loads default settings if no settings are found"""
+    core_version = "stable"
+
+    config = read_config_file()
+
+    try:
+        image = config["core"]["image"]
+        core_version = config["core"]["tag"]
+        binds = config["core"]["binds"]
+        ports = config["core"]["ports"]
+        privileged = config["core"]["privileged"]
+        network = config["core"]["network"]
+
+    except Exception as error:
+        print(f"Error reading startup json data! {error}")
+        sys.exit(1)
+
+    print("Starting core... this could take a while if the image is not available offline...")
+    client.containers.run(
+        f"{image}:{core_version}",
+        name=CORE_CONTAINER_NAME,
+        volumes=binds,
+        ports=ports,
+        privileged=privileged,
+        network=network,
+        detach=True,
+    )
+    print("Started, open http://localhost:8080")
```

Comment:
Can we configure this port somehow ? I believe that 8080 is really complicated since is a default port for a bunch of services, maybe we should pick different number.

Reply:
that is just a text string, that is the port the supervisor used, I'll take it out.